### ISSUE TITLE
[no-extra-parens] False positive when calling generic function with single argument and type parameter with parentheses

### ISSUE THREAD

**Issue by cherryblossom000** [typescript-eslint/typescript-eslint#2314]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/no-extra-parens": 2
  }
}
```

```ts
declare module 'test' {
  class C {}
  export = C
}
declare const f: <T>(x: any) => any

f<import('test')>(1)
```

**Expected Result**

No errors because `f<import('test')>1` isn't valid syntax.

**Actual Result**

```
8:18  error  Unnecessary parentheses around expression  @typescript-eslint/no-extra-parens
```

**Additional Info**

~~This seems to be because of [`getParentSyntaxParen`](https://github.com/mysticatea/eslint-utils/blob/23709f84c8e21f1a38f61ad689c82a8763e0737f/src/is-parenthesized.js#L17-L20) used in [`isParentheszied`](https://github.com/mysticatea/eslint-utils/blob/23709f84c8e21f1a38f61ad689c82a8763e0737f/src/is-parenthesized.js#L10) in eslint-utils. Normally, the `1` in `f(1)` isn't considered to be parenthesised because the `(` to the left of `1` is the same the first `(` after `f`. With `f<import('test')>(1)`, the first `(` after `f` is the one used in `import('test')`, so `1` is reported for being an unnecessarily parenthesised literal.~~

~~This function is used in ESLint in [`isParenthesized`](https://github.com/eslint/eslint/blob/f3a19d2a7ea9505000d14229a450dba133c10d5e/lib/rules/no-extra-parens.js#L167) -> [`hasExcessParensWithPrecedence`](https://github.com/eslint/eslint/blob/f3a19d2a7ea9505000d14229a450dba133c10d5e/lib/rules/no-extra-parens.js#L212) -> [`checkCallNew`](https://github.com/eslint/eslint/blob/f3a19d2a7ea9505000d14229a450dba133c10d5e/lib/rules/no-extra-parens.js#L474).~~

This could be fixed by reducing the precedence of the argument node.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.0` |
| `@typescript-eslint/parser`        | `3.7.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `12.18.2` |
| `npm`                              | `6.14.5` |

**Comment by cherryblossom000** [comment]
This also happens for any type parameter containing parentheses:

```ts
f<(A | B) & C>(1)
```
